### ISSUE TITLE
cosmo-webapp: Not able to perform mkcalendar with latest cosmo-core due to HTTP forbidden method

### ISSUE THREAD

**Issue by pashazz** [1and1/cosmo#25]
So i'm doing a `MKCALENDAR` request and unable to do that because it's not allowed by spring security:

org.springframework.security.web.firewall.RequestRejectedException: The request was rejected because the HTTP method "MKCALENDAR" was not included within the whitelist [HEAD, DELETE, POST, GET, OPTIONS, PATCH, PUT]

As far as I'm concerned this happens with cosmo-core from git (not the release version) and it pulls spring-security-web 5.1.2-RELEASE as dependency.

I'm trying to add my own FilterChainProxy Bean, namely in cosmo-core/src/main/java/org/unitedinternet/cosmo/boot/SecurityFilterConfig I added a new Bean method:

```java

    @Bean
    public FilterChainProxy filterChainProxy () {
        FilterChainProxy proxy = new FilterChainProxy();
        Set<String> httpMethods = new HashSet<>();
        
        httpMethods.add(HttpMethod.DELETE.name());
        httpMethods.add(HttpMethod.GET.name());
        httpMethods.add(HttpMethod.HEAD.name());
        httpMethods.add(HttpMethod.OPTIONS.name());
        httpMethods.add(HttpMethod.PATCH.name());
        httpMethods.add(HttpMethod.POST.name());
        httpMethods.add(HttpMethod.PUT.name());

        // Caldav methods go here
        httpMethods.add("PROPFIND");
        httpMethods.add("PROPPATCH");
        httpMethods.add("COPY");
        httpMethods.add("MOVE");
        httpMethods.add("REPORT");
        httpMethods.add("MKTICKET");
        httpMethods.add("DELTICKET");
        httpMethods.add("ACL");
        httpMethods.add("MKCOL");
        httpMethods.add("MKCALENDAR");

        StrictHttpFirewall firewall = new StrictHttpFirewall();
        firewall.setAllowedHttpMethods(httpMethods);
        proxy.setFirewall(firewall);
        return proxy;
    }
}
```

It was loaded but seems like it's ignored. How do I wire it?

**Comment by danielgrigore** [comment]
Hi @pashazz 
I did some testing with the code above and it worked for me.
From your code I see that you create a `FIlterChainProxy` bean, but to register the filter you need to create a `FilterRegistrationBean` instance.

```
    @Bean
    public FilterRegistrationBean<?> securityFilterChain() {
        ....
        FilterChainProxy filterChainProxy = new FilterChainProxy(filterChain);
        
        Set<String> httpMethods = new HashSet<>();
        
        httpMethods.add(HttpMethod.DELETE.name());
        httpMethods.add(HttpMethod.GET.name());
        httpMethods.add(HttpMethod.HEAD.name());
        httpMethods.add(HttpMethod.OPTIONS.name());
        httpMethods.add(HttpMethod.PATCH.name());
        httpMethods.add(HttpMethod.POST.name());
        httpMethods.add(HttpMethod.PUT.name());

        // Caldav methods go here
        httpMethods.add("PROPFIND");
        httpMethods.add("PROPPATCH");
        httpMethods.add("COPY");
        httpMethods.add("MOVE");
        httpMethods.add("REPORT");
        httpMethods.add("MKTICKET");
        httpMethods.add("DELTICKET");
        httpMethods.add("ACL");
        httpMethods.add("MKCOL");
        httpMethods.add("MKCALENDAR");

        StrictHttpFirewall firewall = new StrictHttpFirewall();
        firewall.setAllowedHttpMethods(httpMethods);
        filterChainProxy.setFirewall(firewall);
        
        FilterRegistrationBean<?> filterBean = new FilterRegistrationBean<>(filterChainProxy);
        filterBean.addUrlPatterns(PATH_DAV);
        return filterBean;        
    }
```
Cheers,

**Comment by danielgrigore** [comment]
Hi @pashazz 
I released a new version of cosmo (3.6.0) which uses a less restrictive HttpFirewall class.
I also added it the possibility to overwrite it. You can have a look at the demo web application that I updated.
Cheers.
